packages/time-series/lib/commands/MRANGE.ts

```diff
@@ -58,10 +60,79 @@ export function transformMRangeArguments(
   return pushGroupByArgument(args, options?.GROUPBY);
 }
 
+export type MrangeRawReplyValue2 = TuplesReply<[
+  key: BlobStringReply,
+  labels: RawLabels2,
+  samples: ArrayReply<Resp2Reply<SampleRawReply>>
+]>;
+
+export type MRangeRawReply2 = ArrayReply<MrangeRawReplyValue2>;
+
+export type MrangeRawReplyValue3 = TuplesReply<[
+  labels: RawLabels3,
+  // TODO: unsure what tod with this element, not part of resp2 at all
+  _: MapReply<BlobStringReply, ArrayReply<unknown>>,
+  samples: ArrayReply<SampleRawReply>
+]>;
+
+export type MrangeRawReplyValueGrouped3 = TuplesReply<[
+  labels: RawLabels3,
+  reducers: MapReply<BlobStringReply, ArrayReply<unknown>>,
+  sources: MapReply<BlobStringReply, ArrayReply<unknown>>,
+  samples: ArrayReply<SampleRawReply>
+]>;
+
+export type MRangeRawReply3 = MapReply<
+  BlobStringReply,
+  MrangeRawReplyValue3 | MrangeRawReplyValueGrouped3
+>;
+
+export interface MRangeReplyItem2 {
+  samples: Array<ReturnType<typeof transformSampleReply[2]>>;
+}
+
+export interface MRangeReplyItem3 {
+  samples: Array<ReturnType<typeof transformSampleReply[3]>>;
+}
+
+export function getSamples(
```

Comment:
im not sure i understand what is going on here.. in resp2 the reply with and without groupby is compatible, but in resp3 its not? (right?)
and the code is just ignoring the new field in resp3? (right?)
i'm confused..

Reply:
yes, the resp2/resp3 responses aren't compatible.  resp2 pushes groupby metadata into the labels (i.e. not really part of labels, but no where else to return them, and hence gets pushed out as labels), while in resp3 they are separate fields in the returned value, that are just ignored (as not labels, and all we do is return labels).